R/comorbidity_index.R

```diff
@@ -184,7 +186,14 @@ comorbidity_index <- function(ipdiag, erdiag, map, weights, at_admission = TRUE,
 
   scores <- score(comorbidities, weights = weights, assign0 = TRUE)
 
-  res <- cbind(comorbidities$genc_id, scores) %>% as.data.table()
+  res <- cbind(genc_id = comorbidities$genc_id, scores) %>% as.data.table()
+
+  if (at_admission) {
+    res <- all_diagnoses %>%
+      select(genc_id) %>%
```

Comment:
Here it should be `select(unique(genc_id))` otherwise there are duplicated rows for each `genc_id` in the output.